### PR TITLE
allow for pvc to be added to multiple workbenches

### PR DESCRIPTION
**Pull request by Gkrumbach07**
https://issues.redhat.com/browse/RHOAIENG-12803

## Description

This PR implements the functionality to allow a PVC to be added to multiple workbenches. The changes include:
- Updating the UI to allow selection of multiple workbenches when adding or editing a PVC.
- Modifying the logic to handle attaching a PVC to multiple workbenches.

I also updated some storage class logic and manifests

https://github.com/user-attachments/assets/a25b7f9a-7bef-4eb4-b5e2-b1531aba3805

## How Has This Been Tested?

1. create two workbenches
   - one with its own new pvc
   - the second one should be able to use the previous pvc
2. go to cluster storage tab
3. create a new pvc and connect one workbench
4. save and re edit pvc and add another workbench 

## Test Impact
- Cypress tests for adding a new PVC and attaching it to multiple workbenches

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by andrewballantyne — COMMENTED** [review]
Looks good to me -- I have not tested it nor reviewed all the code.

One comment about permissions.

**Review by andrewballantyne — COMMENTED** [review]
You have a bug where you're stopping them from any connection path that starts with a prexisting connection path

```
PVC1 - /data
PVC2 - /data2 ERROR << This should not error
```

File: frontend/src/pages/projects/notebook/ConnectedNotebookNames.tsx [hunk 1]
```diff
@@ -32,13 +32,13 @@ const ConnectedNotebookNames: React.FC<ConnectedNotebookNamesProps> = ({
   }
 
   return (
-    <List isPlain>
+    <LabelGroup isCompact>
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/projects/notebook/ConnectedNotebookNames.tsx:35]: I tested on the PF website, not sure if the doc is outdated, but I am pretty sure there is no need to add `isCompact` on `LabelGroup`. There is no hurt to add it though, just a small nit.

File: manifests/core-bases/base/cluster-role.yaml [hunk 2]
```diff
@@ -3,6 +3,15 @@ apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: odh-dashboard
 rules:
+  - verbs:
+      - get
+      - list
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply manifests/core-bases/base/cluster-role.yaml:8]: can they? i will have to test this next week.

**Review by DaoDaoNoCode — APPROVED** [review]
Check the code and tested, other than the comment I left above it lgtm.
/lgtm

**Comment by DaoDaoNoCode** [comment]
@andrewballantyne Should the bug you mentioned above be fixed in my ticket? [RHOAIENG-7680](https://issues.redhat.com/browse/RHOAIENG-7680) It's all about the validation of the storage path, I will post a PR later this afternoon.

**Comment by DaoDaoNoCode** [comment]
/lgtm
As long as https://github.com/opendatahub-io/odh-dashboard/pull/3231#discussion_r1768950814 and https://github.com/opendatahub-io/odh-dashboard/pull/3231#discussion_r1768952289 are addressed next week, this PR is ready to go.
The bug mentioned in https://github.com/opendatahub-io/odh-dashboard/pull/3231#pullrequestreview-2318816652 will be fixed in https://github.com/opendatahub-io/odh-dashboard/pull/3236.

File: manifests/core-bases/base/cluster-role.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply manifests/core-bases/base/cluster-role.yaml]: updated

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Pull request merged**